reaktive/src/commonMain/kotlin/com/badoo/reaktive/single/RepeatUntil.kt

```diff
@@ -0,0 +1,6 @@
+package com.badoo.reaktive.single
+
+import com.badoo.reaktive.observable.Observable
+import com.badoo.reaktive.observable.repeatUntil
+
+fun <T> Single<T>.repeatUntil(predicate: () -> Boolean): Observable<T> = asObservable().repeatUntil(predicate)
```

Comment:
It would be really nice to provide Single result in callback. This is something that is missing in RxJava but would be nice to have. So we will be able to check the result and decide whether we should repeat or not.
So the it should be something like: `repeatUntil(predicate: (T) -> Boolean)`. For `Maybe` is ok as it is now, because `Maybe` may complete without a value and the T is nullable. Looks like a separate implementation would be needed.

Reply:
Done in a separate commit